src/postgres/tree/PostgresServerTreeItem.ts

```diff
@@ -73,4 +76,46 @@ export class PostgresServerTreeItem extends AzureParentTreeItem<ISubscriptionCon
                 return false;
         }
     }
+
+    public async checkAndConfigureFirewall(): Promise<void> {
+        const firewallRuleName: string = 'azureDatabasesForVSCode-publicIp';
+        const resourceGroup: string = azureUtils.getResourceGroupFromId(this.id);
+        const client: PostgreSQLManagementClient = createAzureClient(this.root, PostgreSQLManagementClient);
+        const serverName: string = nonNullProp(this.server, 'name');
+        const ip: string = await publicIp.v4();
+        let existingFirewallRule: FirewallRule | undefined;
+
+        // Creating/updating a firewall rule takes a long time. So check if the rule already exists and is valid before creating/updating
+        try {
+            existingFirewallRule = await client.firewallRules.get(resourceGroup, serverName, firewallRuleName);
+        } catch (error) {
+            if (parseError(error).errorType === 'ResourceNotFound') {
+                // This firewall rule is not configured yet. Ignore error
+            } else {
+                throw error;
+            }
+        }
+
+        if (!existingFirewallRule || existingFirewallRule.startIpAddress !== ip || existingFirewallRule.endIpAddress !== ip) {
+            await ext.ui.showWarningMessage(
```

Comment:
Haha I would phrase this more as "Your current setup is not supported. Would you like to change it?" with a "Yes change it" button instead of a "We're going to change this for you" with a "I guess I'm along for the ride" button

Reply:
I'll wait to change this until https://github.com/microsoft/vscode-cosmosdb/pull/1397 is merged, it'll touch the same code in the database tree item